src/Hill.jl

```diff
@@ -37,25 +36,33 @@ end
 """ Hill optimization function. """
 function optimize_hill(conc::Vector, g1::Matrix, g2::Matrix; maxstep = 300000)
 
+    m = Model(Ipopt.Optimizer)
+    @variables(m, begin
+    1e-5 <= x[1:20] <= 500
+    end
+    )
     f(x) = residHill(x, conc, g1, g2)
 
+    @constraint(m, x[3] <= x[15])
+    @constraint(m, x[4] <= x[16])
+    @constraint(m, x[5] <= x[17])
+    @constraint(m, x[6] <= x[18])
+    @constraint(m, x[7] <= x[19])
+    @constraint(m, x[8] <= x[20])
+    @objective(m, Min, f(x))
     # [EC50, k, max_a1, max_a2, max_b1,  max_b2, max_b3, max_b4, max_g11, max_g12, max_g21, max_g22, max_g23, max_g24, min_a1, min_a2, min_b1,  min_b2, min_b3, min_b4]
-    low = [minimum(conc); 1e-9 * ones(19)]
-    high = [maximum(conc); 10.0; 2.5 * ones(18)]
-
-    return optimize_helper(f, low, high, maxstep)
+    # low = [minimum(conc); 1e-9 * ones(19)]
+    # high = [maximum(conc); 10.0; 2.5 * ones(18)]
+    optimize!(model)
+    println(value(x))
+    # return optimize_helper(f, low, high, maxstep)
 end
 
-function getODEparams(p, conc)
-    if length(p) == 76
-        nMax = 5
-    elseif length(p) == 20
-        nMax = 1
-    elseif length(p) == 34
-        nMax = 2
-    end
+function getODEparams(p, concs)
+
+    nMax = Int((length(p) - 6) / 14)
 
-    effects = zeros(eltype(p), 12, length(conc[:, 1]), Int((length(p) - 6) / 14))
+    effects = zeros(eltype(p), 12, length(conc[:, 1]), nMax)
```

Comment:
I have set it up here, but it has `type` problems that I can't make sense of it. 
This is the error:

![image](https://user-images.githubusercontent.com/34812750/118343707-47c91800-b4df-11eb-83e7-1fdef8e738b2.png)

I also tried fitting for only the control condition, but it threw another `type` error for `startV()`.

Reply:
Why do you need JuMP? I don’t know anything about how to use it.

Reply:
If it’s because Optim.jl doesn’t have the right constraints, I’d start by just adding a penalty in the cost function for the region you want to avoid.

Reply:
Yes, that's why I set this up. Thank you, sure, I will look into that and implement it in this branch.